File: 프로그래머스/0/120852. 소인수분해/소인수분해.java
import java.util.*;
import java.util.stream.*;
class Solution {
    public int[] solution(int n) {
        int[] prime = new int[n + 1];
        for(int i = 2; i <= n; i++){
            if(prime[i] == 1) continue;
            if(i != 2 && i % 2 == 0) prime[i] = 1;
            for(int j = 2*i; j <= n; j += i){
                prime[j] = 1;
            }
        }
        List<Integer> ret = new ArrayList<>();
        for(int i = 2; i <= n; i++){
            if(prime[i] == 0 && n % i == 0) ret.add(i);
        }
        return ret.stream().mapToInt(Integer::intValue).toArray();
    }
}

File: 프로그래머스/0/120852. 소인수분해/소인수분해.py
def solution(n):
    isprime = [True] * (n+1)
    
    for i in range(2, n+1):
        if not isprime[i]: continue
        for j in range(i*2, n+1, i):
            isprime[j] = False

    prime = [x for x in range(2, n+1) if isprime[x]]
    return [x for x in prime if n % x == 0]
        
         
    